src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/OffenderVisitBalanceRepository.kt

```diff
@@ -1,8 +1,26 @@
 package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository
 
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.jpa.repository.Query
 import org.springframework.data.repository.CrudRepository
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderVisitBalance
 
 @Repository
-interface OffenderVisitBalanceRepository : CrudRepository<OffenderVisitBalance, Long>
+interface OffenderVisitBalanceRepository : CrudRepository<OffenderVisitBalance, Long> {
+  @Query(
+    value = """
+      select ovb
+        from OffenderVisitBalance ovb join ovb.offenderBooking ob
+        where ob.bookingSequence = 1
+        and (:prisonId is null or ob.location.id = :prisonId)
+    """,
+    countQuery = """
+      select count(ovb) from OffenderVisitBalance ovb join ovb.offenderBooking ob
+        where ob.bookingSequence = 1 
+        and (:prisonId is null or ob.location.id = :prisonId)
+    """,
+  )
+  fun findForCurrentBooking(prisonId: String? = null, pageable: Pageable): Page<OffenderVisitBalance>
```

Comment:
is this a current booking or latest booking?  Current implies it is active doesn't it?

Reply:
yes - will swap